### DISCUSSION TITLE
Onboarding document for new Rucio core contributors

### DISCUSSION THREAD

**Discussion by rdimaio** [rucio/rucio#6572] [General]
Follow-up from the Rucio Retreat 2024. It might be a good idea to have a document with guidelines and first step for new members of the core Rucio development team.

Here's an initial draft for the doc: https://codimd.web.cern.ch/z7tw-U9YRr2pm7uqg3XLpg (note: CodiMD requires a CERN account, so we might want to put the final version somewhere else)

Feel free to either contribute directly to the doc, or to add suggestions in this thread.

**Comment by voetberg** [comment]
This is fantastic, thank you! 

Suggested additions: 
* Primer on the workflow (issue assignment, making a branch, commit format and PR requirements, the review process) 
* How to use the pre-commits (people who have not used pre-commits may not understand why their commits don't go through, or even that they don't go through in my experience) 
* Differences in development on core/client/api 
* Code cleanliness guidelines, link to general development suggestions (Typing, including documentation, error handling, etc) 

* Link user docs so people have a grasp of how Rucio is used (let them play around with commands) 
* Provide links to the dev docker image
* Provide system requirements for a dev machine (So people can provision a VM if needed) 

* Link the `good first issue` filtered issues in the learning tasks
